src/runtime/capabilities-resolver.ts

```diff
@@ -13,11 +13,43 @@ import {ArcId} from './id.js';
 import {Capabilities} from './capabilities.js';
 import {StorageKey} from './storageNG/storage-key.js';
 import {DriverFactory} from './storageNG/drivers/driver-factory.js';
+import {Schema} from './schema.js';
+import {ReferenceModeStorageKey} from './storageNG/reference-mode-storage-key.js';
 
-export type StorageKeyOptions = Readonly<{
+export type CapabilitiesResolverOptions = Readonly<{
   arcId: ArcId;
 }>;
 
+export abstract class StorageKeyOptions {
+  constructor(public readonly arcId: ArcId,
+              public readonly schemaHash: string,
+              protected readonly schemaName: string = null) {}
```

Comment:
Nit: as far as I am aware, we don't usually indent like this in our TypeScript code.

```
  constructor(
      public readonly arcId: ArcId,
      public readonly schemaHash: string,
      protected readonly schemaName: string = null) {}
```

Reply:
done